code/modules/transport/pods/secondary_system.dm

```diff
@@ -325,6 +328,56 @@
 			break
 	..()
 
+/obj/item/shipcomponent/secondary_system/cargo/jumpseat
+	name = "Living Hold"
+	desc = "Allows the ship to load people as cargo. It is designed to integrate with the Life-Support System so it is probably safe?."
+	acceptable = list(/mob/living/carbon, /mob/living/silicon)
+	maxcap = 1
+	hud_state = "jumpseat"
+
+/obj/item/shipcomponent/secondary_system/cargo/jumpseat/handle_internal_lifeform(mob/lifeform_inside_me, breath_request, mult)
+		. = src.ship.handle_internal_lifeform(lifeform_inside_me, breath_request, mult)
+
+/obj/item/shipcomponent/secondary_system/cargo/jumpseat/activate()
+	var/loadmode = tgui_input_list(usr, "Unload/Load", "Unload/Load", list("Load", "Unload"))
+	switch(loadmode)
+		if("Load")
+			var/atom/movable/AM = null
+			for(var/atom/movable/A in get_step(ship.loc, turn(ship.dir,180) ))
+				if(!A.anchored)
+					AM = A
+					break
+			if(AM)
+				actions.start(new /datum/action/bar/icon/callback(AM, src.ship, 2 SECONDS, /obj/item/shipcomponent/secondary_system/cargo/proc/load, list(AM, usr), \
+																src.icon, src.icon_state, "[AM] successfully enters [his_or_her(AM)] \the [src.ship]'s!", call_proc_on=src), AM)
```

Comment:
Any reason not to use the `SETUP_GENERIC_ACTIONBAR` helper?

Reply:
I thought so, but I'm pretty sure you are right.